x-pack/plugins/alerting/server/routes/find_rules.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { ILicenseState } from '../lib';
+import { FindOptions, FindResult } from '../alerts_client';
+import { RewriteRequestCase, RewriteResponseCase, verifyAccessAndContext } from './lib';
+import { AlertTypeParams, AlertingRequestHandlerContext, BASE_ALERTING_API_PATH } from '../types';
+
+// query definition
+const querySchema = schema.object({
+  per_page: schema.number({ defaultValue: 10, min: 0 }),
+  page: schema.number({ defaultValue: 1, min: 1 }),
+  search: schema.maybe(schema.string()),
+  default_search_operator: schema.oneOf([schema.literal('OR'), schema.literal('AND')], {
+    defaultValue: 'OR',
+  }),
+  search_fields: schema.maybe(schema.arrayOf(schema.string())),
+  sort_field: schema.maybe(schema.string()),
+  sort_order: schema.maybe(schema.oneOf([schema.literal('asc'), schema.literal('desc')])),
+  has_reference: schema.maybe(
+    // use nullable as maybe is currently broken
+    // in config-schema
+    schema.nullable(
+      schema.object({
+        type: schema.string(),
+        id: schema.string(),
+      })
+    )
+  ),
+  fields: schema.maybe(schema.arrayOf(schema.string())),
+  filter: schema.maybe(schema.string()),
+});
+
+const rewriteQueryReq: RewriteRequestCase<FindOptions> = ({
+  default_search_operator: defaultSearchOperator,
+  has_reference: hasReference,
+  search_fields: searchFields,
+  ...rest
+}) => ({
+  ...rest,
+  defaultSearchOperator,
+  ...(hasReference ? { hasReference } : {}),
+  ...(searchFields ? { searchFields } : {}),
+});
+const rewriteBodyRes: RewriteResponseCase<FindResult<AlertTypeParams>> = ({
+  perPage,
+  data,
+  ...restOfResult
+}) => {
+  return {
+    ...restOfResult,
+    per_page: perPage,
+    data: data.map(
+      ({
+        alertTypeId,
+        createdBy,
+        updatedBy,
+        createdAt,
+        updatedAt,
+        apiKeyOwner,
+        notifyWhen,
+        muteAll,
+        mutedInstanceIds,
+        executionStatus: { lastExecutionDate, ...executionStatus },
+        actions,
+        ...rest
+      }) => ({
+        ...rest,
+        rule_type_id: alertTypeId,
+        created_by: createdBy,
+        updated_by: updatedBy,
+        created_at: createdAt,
+        updated_at: updatedAt,
+        api_key_owner: apiKeyOwner,
+        notify_when: notifyWhen,
+        mute_all: muteAll,
+        muted_instance_ids: mutedInstanceIds,
```

Comment:
Should this be `muted_alert_ids` instead?

Reply:
Good catch! Fixed in https://github.com/elastic/kibana/pull/93980/commits/075315503ebd0def045a4cd8004d6a15e2de5084.